include/openssl/crypto.h

```diff
@@ -306,17 +306,20 @@ int CRYPTO_secure_allocated(const void *ptr);
 int CRYPTO_secure_malloc_initialized(void);
 size_t CRYPTO_secure_actual_size(void *ptr);
 size_t CRYPTO_secure_used(void);
+void CRYPTO_get_alloc_counts(int *mcount, int *rcount, int *fcount);
 
 void OPENSSL_cleanse(void *ptr, size_t len);
 
 # ifndef OPENSSL_NO_CRYPTO_MDEBUG
-#  define OPENSSL_mem_debug_push(info) \
-        CRYPTO_mem_debug_push(info, OPENSSL_FILE, OPENSSL_LINE)
-#  define OPENSSL_mem_debug_pop() \
-        CRYPTO_mem_debug_pop()
-int CRYPTO_mem_debug_push(const char *info, const char *file, int line);
-int CRYPTO_mem_debug_pop(void);
-void CRYPTO_get_alloc_counts(int *mcount, int *rcount, int *fcount);
+#  if !OPENSSL_API_3
+#    define OPENSSL_mem_debug_push(info) \
+         CRYPTO_mem_debug_push(info, OPENSSL_FILE, OPENSSL_LINE)
+#    define OPENSSL_mem_debug_pop() \
+         CRYPTO_mem_debug_pop()
+#  endif
+DEPRECATED_IN_3(int CRYPTO_mem_debug_push(const char *info,
+                                          const char *file, int line))
+DEPRECATED_IN_3(int CRYPTO_mem_debug_pop(void))
```

Comment:
DEPRECATEDIN_3

Reply:
yeah, that was in my buffer, not committed.  pushed.

Reply:
What push are you referring to?  It's still not right...